modules/noise/noise_texture.cpp

```diff
@@ -143,18 +155,42 @@ Ref<Image> NoiseTexture::_generate_texture() {
 	Ref<Image> image;
 
 	if (seamless) {
-		image = ref_noise->get_seamless_image(size.x, size.y, invert, seamless_blend_skirt);
+		image = ref_noise->get_seamless_image(size.x, size.y, invert, in_3d_space, seamless_blend_skirt);
 	} else {
-		image = ref_noise->get_image(size.x, size.y, invert);
+		image = ref_noise->get_image(size.x, size.y, invert, in_3d_space);
+	}
+	if (color_ramp.is_valid()) {
+		image = _modulate_with_gradient(image, color_ramp);
 	}
-
 	if (as_normal_map) {
 		image->bump_map_to_normal_map(bump_strength);
 	}
+	if (generate_mipmaps) {
+		image->generate_mipmaps();
+	}
 
 	return image;
 }
 
+Ref<Image> NoiseTexture::_modulate_with_gradient(Ref<Image> p_image, Ref<Gradient> p_gradient) {
+	int width = p_image->get_width();
+	int height = p_image->get_height();
+
+	Ref<Image> new_image;
+	new_image.instantiate();
+	new_image->create(width, height, false, Image::FORMAT_RGBA8);
+
+	for (int row = 0; row < height; row++) {
+		for (int col = 0; col < width; col++) {
+			Color pixel_color = p_image->get_pixel(col, row);
+			Color ramp_color = color_ramp->get_color_at_offset(pixel_color.get_luminance());
```

Comment:
```suggestion
			Color ramp_color = color_ramp->get_color_at_offset(pixel_color.r);
```

Does it need to use luminance here, or can it use the raw noise value? Before this PR it used the raw noise value, whereas `get_luminance()` does some kind of color space conversion.

If you do change this then the code from the next comment will need to be updated, because it includes this line with `get_luminance()`.

Reply:
I think get_luminance calculates the perceived brightness of a pixel, so for a grayscale image where r=g=b the computed value is the same as the r component. However, if a noise library generates an image in color all channels are included with get_luminance().